### PR TITLE
Add mappings for jumping to previous or next git item.

### PR DESCRIPTION
**Pull request by kristijanhusak**
This is one more feature request that i mentioned in #51 .

It's just a simple search for any git icon from current line. This is how I implemented it for defx, and nerdtree git plugin does the same. No one complained so far :)

### CODE REVIEW

File: lua/tree.lua [hunk 1]
```diff
@@ -28,6 +29,21 @@ function M.open()
...
 
+
+local function go_to_git_item(mode)
+  local icon_state = config.get_icon_state()
+  if not icon_state.show_git_icon then return end
+
+  local icons = {}
+  for _, icon in pairs(icon_state.icons.git_icons) do
+    table.insert(icons, icon)
+  end
```
Comment (kyazdani42) [inline lua/tree.lua:40]: this should be cached to improve performance. Also checkout `vim.tbl_values` function.

File: lua/tree.lua [hunk 2]
```diff
@@ -28,6 +29,21 @@ function M.open()
...
+
+  icons = table.concat(icons, '\\|')
+  local flags = mode == 'prev_git_item' and 'b' or ''
```
Comment (kyazdani42) [inline lua/tree.lua:43]: you could make a closure over this variable and load specific functions for each action.
```lua
function gen_go_to(mode)
   local flags = mode == 'prev_git_item' and 'b' or ''
   return function()
    -- uses mode
   end
end

local go_to_prev = gen_go_to('prev_git_item')
local go_to_next = go_to_to()
```

File: lua/tree.lua [hunk 3]
```diff
@@ -55,6 +71,10 @@ function M.on_keypress(mode)
     return lib.toggle_ignored()
   end
 
+  if mode == 'next_git_item' or mode == 'prev_git_item' then
+    return go_to_git_item(mode)
+  end
```
Comment (kyazdani42) [inline lua/tree.lua:76]: and then you could load specific function if it's next or prev.

File: lua/tree.lua (hunk 1 above)
Comment (kristijanhusak) [reply lua/tree.lua]: Done, thanks.

File: lua/tree.lua (hunk 2 above)
Comment (kristijanhusak) [reply lua/tree.lua:34]: Done, thanks.

File: lua/tree.lua (hunk 3 above)
Comment (kristijanhusak) [reply lua/tree.lua:77]: I'm using separate if blocks now to call cached functions.

**Pull request merged**